src/db/user.rs

```diff
@@ -170,14 +228,36 @@ pub async fn update_password(
 
     let updated_fields = doc! {
         "$set":  doc!{
-            "password": _new_password
+            "password": password_hash
         }
     };
 
     let user = collection
         .find_one_and_update(filter, updated_fields, find_one_and_update_options)
-        .await
-        .unwrap();
+        .await?;
 
-    Ok(user)
+    user.ok_or(AppError::CustomError {
+        msg: format!("no user found with id {}", _user_id),
+        code: 500,
+    })
+}
+
+async fn verify_message(addr: &str, sig: &str) -> bool {
+    let message = "Unlock wallet to access nhl-pool-ethereum.".to_string();
+    let signature = hex::decode(sig.strip_prefix("0x").unwrap()).unwrap();
+    let signer_addr = web3::signing::recover(&eth_message(message), &signature[..64], 0).unwrap();
+
+    format!("{:02X?}", signer_addr) == *addr.to_lowercase()
+}
+
+pub fn eth_message(message: String) -> [u8; 32] {
```

Comment:
Cette fonction devrait prendre `&str` en paramètre.

De cette façon tu n'aurais pas à créer un string dans `verify_message`, tu pourrais avoir `const message: &str = "Unlock wallet to access nhl-pool-ethereum."`